aqua/hyper/Portfile

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        vercel hyper 3.0.2
+revision            0
+
+homepage            https://hyper.is
+
+description         A terminal built on web technologies
+
+long_description    {*}${description}. The goal of the project is to create a \
+                    beautiful and extensible experience for command-line \
+                    interface users, built on open web standards. In the \
+                    beginning, our focus will be primarily around speed, \
+                    stability and the development of the correct API for \
+                    extension authors.
+
+categories          aqua shells
+license             MIT
+platforms           darwin
+
+maintainers         {gmail.com:herby.gillot @herbygillot} \
+                    openmaintainer
+
+checksums           ${distname}${extract.suffix} \
+                    rmd160  d0d44d2e4163f9ef3b3ffbcb31d9c8af6111ed0f \
+                    sha256  ef9b9883fe25a003243030b9d7ff09ce3b67512daff3495e85669e85e99fffd2 \
+                    size    1665558
+
+patchfiles          patch-package-json.diff
+
+depends_build       port:python27 \
+                    port:yarn
+
+build.env-append    CSC_IDENTITY_AUTO_DISCOVERY=false
+
+use_configure       no
+
+build {
+    # Set up all JS dependencies
+    system -W ${worksrcpath} "yarn --frozen-lockfile"
```

Comment:
Will this not download dependencies at build time? This could lead to a similar issue to https://trac.macports.org/ticket/61192.

I don't have any experience with JS portfiles, so please correct me if I'm wrong on this.

Reply:
You're a 100% right, but we don't have a solution for pre-download Node modules the way we do for Go, at least as far as I'm aware.  It's currently this way for a number of other ports, such as `beekeeper-studio`, `Zettlr` and a few more.